src/gsy_e/models/strategy/state/heatpump_all_tanks_state.py

```diff
@@ -1,22 +1,32 @@
-import logging
+from logging import getLogger
 from statistics import mean
-from typing import Dict, Union, List
+from typing import Union, List, Dict
 
 from gsy_framework.constants_limits import GlobalConfig, FLOATING_POINT_TOLERANCE
-from gsy_framework.utils import convert_kJ_to_kWh, convert_kWh_to_kJ
+from gsy_framework.utils import convert_kJ_to_kWh
 from pendulum import DateTime
 
-from gsy_e.models.strategy.state.heat_pump_state import TankParameters, HeatPumpTankState
+from gsy_e.models.strategy.state.heatpump_pcm_tank_state import PCMTankState
+from gsy_e.models.strategy.state.heatpump_water_tank_state import HeatPumpTankState
+from gsy_e.models.strategy.energy_parameters.heatpump.tank_parameters import (
+    TankParameters,
+    HeatpumpTankTypes,
+)
 
-logger = logging.getLogger(__name__)
+log = getLogger(__name__)
 
 
 class AllTanksState:
     """Manage the operation of heating and extracting temperature from multiple tanks."""
 
     def __init__(self, tank_parameters: List[TankParameters]):
         self._tanks_states = [
-            HeatPumpTankState(tank, GlobalConfig.slot_length) for tank in tank_parameters
+            (
+                HeatPumpTankState(tank_parameters=tank, slot_length=GlobalConfig.slot_length)
+                if tank.type == HeatpumpTankTypes.WATER
+                else PCMTankState(tank_parameters=tank)
+            )
+            for tank in tank_parameters
         ]
```

Comment:
Minor, but I think I prefer this in order to get warned if the enum selection is not supported:
```
if tank.type == HeatpumpTankTypes.WATER:
    state = HeatPumpTankState(tank_parameters=tank, slot_length=GlobalConfig.slot_length)
elif tank.type == HeatpumpTankTypes.PCM:
    state = PCMTankState(tank_parameters=tank)
else:
   assert False, f"Unsupported heat pump tank type {tank.type}"

```